commands/instance/search.go

```diff
@@ -0,0 +1,65 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package instance
+
+import (
+	"regexp"
+
+	"github.com/urfave/cli"
+
+	"github.com/apache/skywalking-cli/commands/flags"
+	"github.com/apache/skywalking-cli/commands/interceptor"
+	"github.com/apache/skywalking-cli/commands/model"
+	"github.com/apache/skywalking-cli/display"
+	"github.com/apache/skywalking-cli/graphql/client"
+	"github.com/apache/skywalking-cli/graphql/schema"
+)
+
+var SearchCommand = cli.Command{
+	Name:  "search",
+	Usage: "Filter the instance from the existing service instance list",
+	Flags: append(flags.DurationFlags, append(flags.SearchRegexFlags, flags.InstanceServiceIDFlags...)...),
+	Before: interceptor.BeforeChain([]cli.BeforeFunc{
+		interceptor.DurationInterceptor,
+	}),
+	Action: func(ctx *cli.Context) error {
+		serviceID := verifyAndSwitchServiceParameter(ctx)
+
+		end := ctx.String("end")
+		start := ctx.String("start")
+		step := ctx.Generic("step")
+
+		regex := ctx.String("regex")
+
+		instances := client.Instances(ctx, serviceID, schema.Duration{
+			Start: start,
+			End:   end,
+			Step:  step.(*model.StepEnumValue).Selected,
+		})
+
+		var result []schema.ServiceInstance
```

Comment:
This could be not initialized if the instance list is empty. Do you test what will happen for `#Display`?

Reply:
condition: ``` result == nil || len(result) == 0```

+ --display=table 
```
null
```
+ --display=yaml
```
[]
```
+ --display=table
```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/apache/skywalking-cli/display/table.Display(0x13f9c80, 0x18292e0, 0x7ffeefbff494, 0x5)
```

Reply:
> * --display=json
> 
> ```
> null
> ```

Need to replace ```null``` with ```{}```？ @wu-sheng  @kezhenxu94

Reply:
At least, not an exception, like in table mode.

Reply:
I think it's ok to output `null` in JSON format, because `null` is valid in JSON, the `table` style indeed needs to be fixed

Reply:
Is this still unchanged?

Reply:
Added empty collection check to ```Tabld#Display```